### DISCUSSION TITLE
Grove - Capacitive Moisture Sensor (Corrosion Resistant) | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#489] [Q&A]
# Grove - Capacitive Moisture Sensor (Corrosion Resistant) | Seeed Studio Wiki

Grove - Capacitive Moisture Sensor (Corrosion Resistant)

https://wiki.seeedstudio.com/Grove-Capacitive_Moisture_Sensor-Corrosion-Resistant/

**Comment by MihaMarkic** [comment]
Would you guys mind exporting STEP 3D model (or STL if possible) from Altium designer, which is the only choice at the moment (if I'm not mistaken)?

**Reply by xzxcessarr** [reply]
Hi, thanks for feedback! We provide the Eagle files here: https://files.seeedstudio.com/wiki/Grove-Capacitive_Moisture_Sensor_Corrosion_Resistant/res/Grove-Capacitive%20Moisture%20Sensor%20(Corrosion%20Resistant).zip, you can download it and convert it into AD Files add your 3D electronic component to build your stl model. If well you can use this online EDA tool: https://lceda.cn/editor to convert and build stl model by select the encapsulation you need, it is very easy to use.

**Comment by Yonik94** [comment]
Hi, I'm using this sensor with libre le potato computer board (compatible with raspberry pi model 3 B+), with A/D converter MCP3002 (10bit) to get the analog values from the sensor.
It looks strange for me. I'm getting about 100.97-101 voltage when it totally dry and   100.4 when it totally wet (in a water).
Is it make sense? it looks me too small difference.. please help me to understand my mistake if I have it..

# Here is my python code only for the test:
    import time
    import spidev

    spi_ch = 0

    # Enable SPI
    spi = spidev.SpiDev(0, spi_ch)
    spi.max_speed_hz = 1200000

    def read_adc(adc_ch, vref = 3.3):

        # Make sure ADC channel is 0 or 1
        if adc_ch != 0:
            adc_ch = 1

        # Construct SPI message
        # # First bit (Start): Logic high (1)
        # Second bit (SGL/DIFF): 1 to select single mode
        # Third bit (ODD/SIGN): Select channel (0 or 1)
        # Fourth bit (MSFB): 0 for LSB first
        #  Next 12 bits: 0 (don't care)
        msg = 0b11
        msg = ((msg << 1) + adc_ch) << 5
        msg = [msg, 0b00000000]
        reply = spi.xfer2(msg)

        # Construct single integer out of the reply (2 bytes)
        adc = 0
        for n in reply:
            adc = (adc << 8) + n

        # Last bit (0) is not part of ADC value, shift to remove it
        adc = adc >> 1

        # Calculate voltage form ADC value
        voltage = (vref * adc) / 1024

        return voltage

    # Report the channel 0 and channel 1 voltages to the terminal
    try:
        while True:
            adc_0 = read_adc(0)
            adc_1 = read_adc(1)
            print(
                "Ch 0:", round(adc_0, 2),
                "V Ch 1:", round(adc_1, 2),
                "V")
            time.sleep(1)

    finally:
        spi.close()
